spyder/config/user.py

```diff
@@ -5,369 +5,498 @@
+        path = self._path
+        config_file = osp.join(path, '{}.ini'.format(self._name))
+        return config_file
 
     def set_defaults(self, defaults):
+        """Set default values and save to defaults folder location."""
         for section, options in defaults:
             for option in options:
-                new_value = options[ option ]
-                self._set(section, option, new_value, False)
+                new_value = options[option]
+                self._set(section, option, new_value, verbose=False)
 
 
-#==============================================================================
+# ============================================================================
 # User config class
-#==============================================================================
+# ============================================================================
 class UserConfig(DefaultsConfig):
     """
-    UserConfig class, based on ConfigParser
-    name: name of the config
-    defaults: dictionnary containing options
-              *or* list of tuples (section_name, options)
-    version: version of the configuration file (X.Y.Z format)
-    subfolder: configuration file will be saved in %home%/subfolder/%name%.ini
-
-    Note that 'get' and 'set' arguments number and type
-    differ from the overriden methods
+    UserConfig class, based on ConfigParser.
+
+    Parameters
+    ----------
+    name: str
+        Name of the config
+    path: str
+        Configuration file will be saved in path/%name%.ini
+    defaults: {} or [(str, {}),]
+        dictionnary containing options *or* list of tuples (sec_name, options)
+    load: bool
+        TODO:
+    version: str
+        version of the configuration file in 'major.minor.micro' format.
+    backup: bool
+        TODO:
+    raw_mode: bool
+        TODO:
+    remove_obsolete: bool
+        TODO:
+
+    Notes
+    -----
+    The 'get' and 'set' arguments number and type differ from the overriden
+    methods. 'defaults' is an attribute and not a method.
     """
     DEFAULT_SECTION_NAME = 'main'
-    def __init__(self, name, defaults=None, load=True, version=None,
-                 subfolder=None, backup=False, raw_mode=False,
-                 remove_obsolete=False):
-        DefaultsConfig.__init__(self, name, subfolder)
-        self.raw = 1 if raw_mode else 0
-        if (version is not None and
-                re.match(r'^(\d+).(\d+).(\d+)$', version) is None):
-            raise ValueError("Version number %r is incorrect - must be in X.Y.Z format" % version)
-        if isinstance(defaults, dict):
-            defaults = [ (self.DEFAULT_SECTION_NAME, defaults) ]
-        self.defaults = defaults
-        if defaults is not None:
-            self.reset_to_defaults(save=False)
-        fname = self.filename()
-        if backup:
-            try:
-                shutil.copyfile(fname, "%s.bak" % fname)
-            except IOError:
-                pass

```

Comment:
Why a backup is not performed at this point?

Reply:
I put it at the end of the operation but yes, it should be done before the load. Moved back up!